### PR TITLE
Modify the Makefile to use the current VENV or create one if not in a VENV

### PR DESCRIPTION
**Pull request by Natim**
This let the Makefile use the current VENV or create one locally if the user is not inside a VENV.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,31 +1,40 @@
-DEV_STAMP=.dev_env_installed.stamp
-INSTALL_STAMP=.install.stamp
+VIRTUALENV=virtualenv
+VENV := $(shell echo $${VIRTUAL_ENV-.})
```
Comment (Natim) [inline Makefile]: This use the current VENV or default to .

File: Makefile [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
+PYTHON=$(VENV)/bin/python
+DEV_STAMP=$(VENV)/.dev_env_installed.stamp
```
Comment (Natim) [inline Makefile:3]: We create the stamp for each VENV so that we can have multiple VENV and be sure that they are ready.

File: Makefile (hunk 1 above)
Comment (almet) [inline Makefile]: wouldn't it be better to default to .venv or something like that? so we don't clutter the current dir?

Comment (Natim) [inline Makefile]: Yes definitly :+1:

File: Makefile [hunk 3]
```diff
@@ -1,31 +1,39 @@
...
 install-dev: $(DEV_STAMP)
-$(DEV_STAMP):
-	pip install -r dev-requirements.txt
+$(DEV_STAMP): $(PYTHON)
+	$(VENV)/bin/pip install -r dev-requirements.txt
 	touch $(DEV_STAMP)
 
+virtualenv: $(PYTHON)
+$(PYTHON):
+	$(VIRTUALENV) $(VENV)
```
Comment (almet) [inline Makefile]: why not put directly "virtualenv" here and use a variable?

**Pull request merged**